files/aws_lifecycle_hooks/attach_ebs_volume.py

```diff
@@ -59,21 +59,70 @@ def attach_volume(
             raise
 
 
+def parse_user_data():
+    # user_data = tools.get_user_data()
+    return 'a', 'b'
+
+
+def try_attach(
+        volume_id: str,
+        device_name: str,
+        instance_id: str,
+        region: str,
+        retry_limit: int,
+        retry_interval: int,
+):
+    print("""\
+        volume_id {volume_id}
+        device_name {device_name}
+        instance_id {instance_id}
+        region {region}
+        retry_limit {retry_limit}
+        retry_interval {retry_interval}
+    """.format(
+        volume_id=volume_id,
+        device_name=device_name,
+        instance_id=instance_id,
+        region=region,
+        retry_limit=retry_limit,
+        retry_interval=retry_interval,
+    ))
+
+    attached = False
+    retry = 0
+    while not attached:
+        try:
+            retry = retry + 1
+            attach_volume(
+                volume_id=volume_id,
+                device_name=device_name,
+                region_name=region,
+                instance_id=instance_id,
+            )
+            attached = True
+        except VolumeInUse:
```

Comment:
According to PEP-8, the names of Exception classes should end with `Error`

> Because exceptions should be classes, the class naming convention applies here. However, you should use the suffix "Error" on your exception names (if the exception actually is an error).